tests/php/Control/HTTPStreamResponseTest.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+namespace SilverStripe\Control\Tests;
+
+use SilverStripe\Control\HTTPStreamResponse;
+use SilverStripe\Dev\SapphireTest;
+
+class HTTPStreamResponseTest extends SapphireTest
+{
+    /**
+     * Test replaying of stream from memory
+     */
+    public function testReplayStream()
+    {
+        $path = __DIR__ . '/HTTPStreamResponseTest/testfile.txt';
+        $stream = fopen($path, 'r');
+        $response = new HTTPStreamResponse($stream, filesize($path));
+
+        // Test body (should parse stream directly into memory)
+        $this->assertEquals("Test output\n", $response->getBody());
+
+        // Test stream output
+        ob_start();
+        $response->output();
+        $result = ob_get_contents();
+        ob_end_clean();
+
+        $this->assertEquals(12, $response->getHeader('Content-Length'));
+        $this->assertEquals("Test output\n", $result);
+    }
+
+    /**
+     * Test stream directly without loading into memory
+     */
+    public function testDirectStream()
+    {
+        $path = __DIR__ . '/HTTPStreamResponseTest/testfile.txt';
+        $stream = fopen($path, 'r');
+        $response = new HTTPStreamResponse($stream, filesize($path));
+
+        // Test stream output
+        ob_start();
+        $response->output();
+        $result = ob_get_contents();
+        ob_end_clean();
+
+        $this->assertEquals(12, $response->getHeader('Content-Length'));
+        $this->assertEquals("Test output\n", $result);
+        $this->assertEmpty($response->getSavedBody(), 'Body of seekable stream is un-cached');
+
+        // Seekable stream can be repeated
```

Comment:
I couldn't figure out how to mock an unseekable stream, but I will add the suggested assertion.

If the stream is seekable, the test is to ensure it doesn't unnecessarily cache the body on a second invocation.